### ISSUE TITLE
Exception gets logged twice if message isn't specified

### ISSUE THREAD

**Issue by PrometheusMMIV** [NLog/NLog#5272]
NLog version: 4.7.6 (but also reproduced on 5.2.0)
Platform: .Net 4.8

NLog.config
```
<nlog>
  <targets>
    <target name="logFile" type="file" fileName="${basedir}/logs/${shortdate}.log" 
            layout="${longdate} ${message} ${exception:format=tostring}" />
  </targets>

  <rules>
    <logger name="*" minLevel="Info" writeTo="logFile" />
  </rules>
</nlog>
```

Code
```
Logger _logger = LogManager.GetCurrentClassLogger();

var ex = new Exception("THIS IS A TEST ERROR");
_logger.Error(ex);
```

- What is the current result?

Exception is logged twice when a message isn't specified
```2023-06-30 10:53:44.2522 System.Exception: THIS IS A TEST ERROR System.Exception: THIS IS A TEST ERROR```

When passing a message with the exception, the exception is only logged once, as expected
```_logger.Error(ex, "An error occurred.");```
```2023-06-30 10:56:23.3132 An error occurred. System.Exception: THIS IS A TEST ERROR```

- What is the expected result?

When a message is not specified, the behavior should be consistent and only log the exception once 
```2023-06-30 10:53:44.2522 System.Exception: THIS IS A TEST ERROR```

- Are there any workarounds?

You can pass an empty string as the message, but this is not ideal to have to pass an unnecessary value every place where I only want to log the exception with no message.

**Comment by PrometheusMMIV** [comment]
Looking at the method signatures, it looks like there is a method for logging an exception with a message
```public void Error(Exception exception, string message);```

However, there isn't actually a method for logging an exception without a message. Instead, when passing an exception by itself, it is calling this method
```public void Error<T>(T value);```

I would guess that the method above is treating the exception as the object to be logged, and also as an exception, which is causing it to be logged twice.

I believe this could be resolved by adding a method which only accepts an exception parameter and defaults to an empty message, instead of having the exception get treated as a generic value.
```public void Error(Exception exception);```

**Comment by snakefoot** [comment]
What happens if you replace `${message} ${exception:format=tostring}` with `${message:withException=true:exceptionSeparator= }` ?

**Comment by PrometheusMMIV** [comment]
```${message:withException=true}``` does only log the exception once, and it's what I'm planning to use as a workaround for the meantime. However, for those who may have more complex error logging needs (e.g. Data, etc.) the duplication issue is still reproducible in the latest version.

**Comment by snakefoot** [comment]
I like that `${message}` and `${exception:format=tostring}` both produce output when doing `logger.Error(exception)`.

In the early days then both `${message}` and `${exception:format=tostring}` would output full exception details when calling `logger.Error(exception)`. Now `${message}` will "only" produce Exception-Type + Exception-Message (without StackTrace).

**Comment by PrometheusMMIV** [comment]
Yes, in version 4.0 which we're using now, I noticed that it is logging the entire stack trace twice, which is what brought the issue to my attention.

> I like that ${message} and ${exception:format=tostring} both produce output when doing logger.Error(exception).

What would be the purpose of that? If I'm not passing a message, I wouldn't expect it to log a message or duplicate the exception as the message, since that's redundant.

**Comment by snakefoot** [comment]
Well most people expect that `${message}` should include some kind of output when writing a logevent. Preferable something human readable.

If you don't like the default behavior of `${message}` then you can always register your own custom layout-renderer: https://github.com/NLog/NLog/wiki/How-to-write-a-custom-layout-renderer

**Comment by PrometheusMMIV** [comment]
I can't speak for everyone, but in my opinion that seems unintuitive and not desirable. Especially when using more complex layouts besides the default, which is something I would like to look into in the future, but I could see this being an issue.

And just for reference, there was at least one other person who had the same complaint a while back:
https://stackoverflow.com/questions/33613971/nlog-logs-error-twice-in-same-file

**Comment by snakefoot** [comment]
NLog Logger-method `Debug<T>(T obj)` is shorthand for `Debug("{0}", obj)`.

This means the `${message}` output becomes `string.Format("{0}", obj)` or `obj.ToString()`, which is the original behavior.

NLog have over time introduced special optimizations for `Exception`-objects:
- NLog 4.1 - [Assign NLog LogEventInfo.Exception so ${exception} works][1] - Introduced the double output issue
- NLog 4.5.4 - [Assign NLog LogEventInfo.Exception so ${exception} works in more cases][2]
- NLog 4.6 - [Changed ${message:withException=true} to only output exception once][3]
- NLog 5.0 - [Changed ${message} to only output ex.GetTyp() + ex.Message][4]

  [1]: https://github.com/NLog/NLog/pull/788
  [2]: https://github.com/NLog/NLog/pull/2586
  [3]: https://github.com/NLog/NLog/pull/2978
  [4]: https://github.com/NLog/NLog/pull/3684

**Comment by snakefoot** [comment]
Closing issue as answered due to inactivity.
